hooks/use-css-vars/src/generate-tokens-map.ts

```diff
@@ -0,0 +1,35 @@
+import type { Theme, Value } from './use-css-vars.types';
+
+const isObject = (value: Value) =>
+  !!(
+    value &&
+    typeof value === 'object' &&
+    typeof value !== 'string' &&
+    !Array.isArray(value)
+  );
+
+const isEmptyObject = (obj: Value) => {
+  if (typeof obj === 'object' && obj !== null && Object.keys(obj).length !== 0) {
+    return false;
+  } else {
+    return true;
+  }
+};
+
+export function generateTokensMap<T>(theme: Record<string, Value>, prefix?: string): T {
```

Comment:
Do we want to call this something like `flattenIntoTokenMap`?

Reply:
The reason I picked `generate` is because the consumer would use it to generate their token maps. I thought it would be more clear to them that way and they wouldn't need to know the underlying mechanics: just that they are putting in their tokens and getting their token map out. What do you think? Do you think it makes sense?

P.S. I should add a usage example there like in `useCSSVars` to make it clearer.

Reply:
Added a JSDoc with usage to this util! Let me know what you think.